spyder/api/plugins.py

```diff
@@ -815,9 +817,12 @@ def __init__(self, parent, configuration=None):
             # Widget setup
             # ----------------------------------------------------------------
             try:
-                container._setup(options=options)
-            except AttributeError:
-                pass
+                _setup = getattr(container, "_setup", None)
+                if _setup:
+                    _setup(options=options)
+            except AttributeError as error:
```

Comment:
Since above we are using `getattr` and doing a validation for `_setup` is this except reachable? Also, is there any specific case for `_setup` to not be defined? Maybe documenting that here could be useful

Reply:
Good catch. Will fix.

Reply:
Thanks, and about the `_setup` is there any specific situation where this happens?

Reply:
Cannot recall now why I left this. Will remove it.